src/BTrees/_datatypes.py

```diff
@@ -359,22 +385,129 @@ class Q(_AbstractUIntDataType):
     using64bits = True
 
 
-class Bytes(KeyDataType):
+class _AbstractBytes(KeyDataType):
     """
     An exact-length byte string type.
+
+    This must be subclassed to provide the actual byte length.
     """
     __slots__ = ()
-    prefix_code = 'fs'
+    tree_size = 500
     default_bucket_size = 500
 
     def __init__(self, length):
-        super(Bytes, self).__init__()
+        super(_AbstractBytes, self).__init__()
         self._length = length
 
     def __call__(self, item):
         if not isinstance(item, bytes) or len(item) != self._length:
-            raise TypeError("%s-byte array expected" % self._length)
+            raise TypeError("%s-byte array expected, not %r" % (self._length, item))
         return item
 
     def supports_value_union(self):
         return False
+
+
+class f(_AbstractBytes):
+    """
+    The key type for an ``fs`` tree.
+
+    This is a two-byte prefix of an overall 8-byte value
+    like a ZODB object ID or transaction ID.
+    """
+
+    # Our keys are treated like integers; the module
+    # implements IIntegerObjectBTreeModule
+    long_name = 'Integer'
+    prefix_code = 'f'
+
+    def __init__(self):
+        _AbstractBytes.__init__(self, 2)
+
+    def supports_value_union(self):
+        # We don't implement 'multiunion' for fsBTree.
+        return False
+
+    # Check it can be converted to a two-byte
+    # value. Note that even though we allow negative values
+    # that can break test assumptions: -1 < 0 < 1, but the byte
+    # values for those are \xff\xff > \x00\x00 < \x00\x01.
+    _as_2_bytes = Struct('>h').pack
+
+    def coerce(self, item):
+        try:
+            return self(item)
+        except TypeError:
+            try:
+                return self._as_2_bytes(object_to_index(item))
+            except struct_error as e:
+                raise TypeError(e)
+
+    def make_toString(self):
+        def toString(self):
+            return b''.join(self._keys) + b''.join(self._values)
```

Comment:
On the contrary, it's a **method** so `self` is the correct parameter name. In addition, using `self` enables linters to see that, yes, in fact, it's ok that this function is using private attributes (`_keys`, `_values`) of the paramater.

Reply:
> It will become a method - at this place, it is only a function: I had some difficulties to understand `self._keys` and `self._values` which make only sense in the context of a tree/bucket and not in that of a datatype. It took me some moments to recognize that those references to `self` are not the `self` of the defined method.

I've made the builder functions `@staticmethod` so there's no more `self` paramater.